gradle/forbidden-dependencies.gradle

```diff
@@ -0,0 +1,23 @@
+
+// we do not want any of these dependencies on the compilation classpath
+// because they could then be used within Elasticsearch
+List<String> FORBIDDEN_DEPENDENCIES = [
+  'guava'
+]
+
+Closure checkDeps = { Configuration configuration -> 
+  configuration.resolutionStrategy.eachDependency {
+    String artifactName = it.target.name
+    if (FORBIDDEN_DEPENDENCIES.contains(artifactName)) {
+      throw new GradleException("Dependency '${artifactName}' on configuration '${configuration.name}' is not allowed. " +
```

Comment:
We might want to include the full path to the configuration here as just saying "compileClasspath" isn't going to be particularly helpful.

Reply:
This is the current failure output. It has the full path in the first part of the error.

```
> Could not resolve all files for configuration ':x-pack:plugin:identity-provider:compileClasspath'.
   > Could not resolve com.google.guava:guava:19.0.
     Required by:
         project :x-pack:plugin:identity-provider
      > Dependency 'guava' on configuration 'compileClasspath' is not allowed. If it is needed as a transitive depenency, try adding it to the runtime classpath
```